### ISSUE TITLE
Crossplane provider

### ISSUE THREAD

**Issue by eremid** [vmware/terraform-provider-vcd#966]
Hello,

We are working on a Crossplane provider built with Upjet framework. I’m just start testing and I’ve some issues to share with you like :
* string parameters in documentation but bool in provider code.
* remove unused parameters like show_upload_progress in documentation (or adding it in provider code ?).
Examples :
* https://github.com/FrangipaneTeam/terraform-provider-vcd/commit/6ba2e07836b515bd954252034631a63584b44e42
* https://github.com/FrangipaneTeam/terraform-provider-vcd/commit/dd2b8041113f319cb989d17676ea0f1693449dfb

Reading your Code Contribution Flow, this issue is open to ask if it is possible to contribute towards the goal of having a working crossplane provider and how we can contribute ?

Thanks

**Comment by dataclouder** [comment]
Hi
Thanks for your interest in our project.
Contributions are welcome. However, when it comes to documentation, we would like to keep the contributions to the matter of fixing errors.
Thus, removing an unneeded parameter in an example is a very useful fix, but changing every `*` to `-` or `_` is not, as it doesn't solve any problems, and it would make the rest of our documentation inconsistent.

Similarly, changing `"true"` to `true` is not necessary. Although stylistically it's better to use the unquoted `true` or `false`, Terraform doesn't complain and will [automatically convert from Boolean to string when needed](https://developer.hashicorp.com/terraform/language/expressions/types#type-conversion).
Should such a change become necessary in the future, we would prefer changing all of them at once using our own coding tools.

Summing up: if there are changes that affect the usefulness of the documentation, by all means do it, but if you feel suggesting formatting enhancements, be aware that we may not be able to dedicate review time to large stylistic changes with hundreds of lines involved.

WRT 
> it is possible to contribute towards the goal of having a working crossplane provider 

please explain what you have in mind, and how the current `terraform-provider-vcd` should be involved.
In other words, if the current code could be enhanced to be used in a crossplane provider, please outline what such enhancements should be.

**Comment by eremid** [comment]
> Contributions are welcome. However, when it comes to documentation, we would like to keep the contributions to the matter of fixing errors.

OK.

> Similarly, changing "true" to true is not necessary.

It’s better for the crossplane provider, I detail this point below.

I’m using [upjet](https://github.com/upbound/upjet) code generator to build the crossplane provider. The crossplane APIs and kubernetes CRD generated from the terraform provider code missmatch with the example generated (based on the terraform provider documentation). 
For example :
* The Kubernetes CRD with `cacheEnabLed` and `deleteForce` :
```
---
apiVersion: apiextensions.k8s.io/v1
kind: CustomResourceDefinition
metadata:
  annotations:
    controller-gen.kubebuilder.io/version: v0.8.0
  creationTimestamp: null
  name: catalogs.catalog.vcd.upbound.io
spec:
  group: catalog.vcd.upbound.io
  names:
    categories:
    - crossplane
    - managed
    - vcd
    kind: Catalog
    listKind: CatalogList
    plural: catalogs
    singular: catalog
  scope: Cluster
  versions:
  - additionalPrinterColumns:
    - jsonPath: .status.conditions[?(@.type=='Ready')].status
      name: READY
      type: string
    - jsonPath: .status.conditions[?(@.type=='Synced')].status
      name: SYNCED
      type: string
    - jsonPath: .metadata.annotations.crossplane\.io/external-name
      name: EXTERNAL-NAME
      type: string
    - jsonPath: .metadata.creationTimestamp
      name: AGE
      type: date
    name: v1beta1
    schema:
      openAPIV3Schema:
        description: Catalog is the Schema for the Catalogs API. Provides a VMware
          Cloud Director catalog resource. This can be used to create and delete a
          catalog.
        properties:
          apiVersion:
            description: 'APIVersion defines the versioned schema of this representation
              of an object. Servers should convert recognized schemas to the latest
              internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources'
            type: string
          kind:
            description: 'Kind is a string value representing the REST resource this
              object represents. Servers may infer this from the endpoint the client
              submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds'
            type: string
          metadata:
            type: object
          spec:
            description: CatalogSpec defines the desired state of Catalog
            properties:
              deletionPolicy:
                default: Delete
                description: DeletionPolicy specifies what will happen to the underlying
                  external when this managed resource is deleted - either "Delete"
                  or "Orphan" the external resource.
                enum:
                - Orphan
                - Delete
                type: string
              forProvider:
                properties:
                  cacheEnabled:
                    description: Enable early catalog export to optimize synchronization.
                      Default is false. It is recommended to set it to true when publishing
                      the catalog. True enables early catalog export to optimize synchronization
                    type: boolean
                  deleteForce:
                    description: When destroying use delete_force=true with delete_recursive=true
                      to remove a catalog and any objects it contains, regardless
                      of their state When destroying use delete_force=True with delete_recursive=True
                      to remove a catalog and any objects it contains, regardless
                      of their state.
                    type: boolean
[...]
```

* The example generated from the documentation :
```
apiVersion: catalog.vcd.upbound.io/v1beta1
kind: Catalog
metadata:
  annotations:
    meta.upbound.io/example-id: catalog/v1beta1/catalog
  labels:
    testing.upbound.io/example-name: myNewCatalog
  name: mynewcatalog
spec:
  forProvider:
    deleteForce: "true"
    deleteRecursive: "true"
    description: catalog for files
    name: my-catalog
    org: my-org

---
```

This is the reason behind the string to bool change in the documentation.

Same issue in `vcd_catalog_vapp_template` with `show_upload_progress` present in the documentation but not in the terraform code : in my crossplane provider, I’ve got an example with `showUploadProgress: true` but nothing in the CRD.

I hope that clarifies things 🙂

**Comment by dataclouder** [comment]
I understand why you want to change the quoted boolean values. Even though it is not of immediate value to us, in the end it will be an useful clean-up.
Thus, if you submit a pull request to change all `"true"` into `true` and all `"false"` into `false` within the documentation, we will probably accept it. 
Please consider, however, that we may not accept –or perhaps even review– a pull request that contain dozens of `*` replaced by `-` or `_` in markdown files. It is not only a matter of style preference: this documentation gets parsed by Hashicorp registry publisher to generate the provider official documentation, and so far this is the syntax that works. We won't change it unless we know for sure that the registry parser **requires** it.

As for the `show_upload_progress` in `vcd_catalog_vapp_template`, I will fix it, as it is a mistake, and I am cleaning up that part of the code in a bug fix that will be released in January.
